apps/dapp-console/app/settings/contracts/StartVerificationDialog.tsx

```diff
@@ -0,0 +1,246 @@
+import { LONG_DURATION } from '@/app/constants/toast'
+import { shortenAddress } from '@eth-optimism/op-app'
+import {
+  Button,
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+  Input,
+  Label,
+  Textarea,
+  useToast,
+} from '@eth-optimism/ui-components'
+import { Text } from '@eth-optimism/ui-components/src/components/ui/text/text'
+import {
+  RiArrowLeftLine,
+  RiCheckboxCircleFill,
+  RiFileCopyFill,
+} from '@remixicon/react'
+import { useCallback, useMemo, useState } from 'react'
+import { Address, isHash } from 'viem'
+
+export type StartVerificationDialogStep =
+  | 'begin'
+  | 'start-verification'
+  | 'finish-verification'
+  | 'verified'
+
+export type StartVerificationContentProps = {
+  contractAddress?: Address
```

Comment:
it would always exist i just had it as an optional to make rendering the UI easier, i think some of the types will start to adjust slightly as we integrate with the backend